framework/src/timeintegrators/CrankNicolson.C

```diff
@@ -76,14 +76,20 @@ CrankNicolson::postResidual(NumericVector<Number> & residual)
 {
   // PETSc 3.19 insists on having closed vectors when doing VecAXPY,
   // and that's probably a good idea with earlier versions too, but
-  // we don't always get here with _Re_time closed.  close() is
-  // slow but a closed() check is fast and embarrassingly
-  // parallel, so let's just test them all.
-  if (!_Re_time.closed())
+  // we don't always get here with _Re_time closed.
+  std::vector<unsigned char> inputs_closed = {
+      _Re_time.closed(), _Re_non_time.closed(), _residual_old.closed()};
+
+  // We might have done work on one processor but not all processors,
+  // so we have to sync our closed() checks.  Congrats to the BISON
+  // folks for test coverage that caught that.
+  comm().min(inputs_closed);
+
+  if (!inputs_closed[0])
```

Comment:
This is not a required change, but we would also support a `min` with a `std::tuple` right? (Then you could do an explicit structured binding). Mostly for confirmation/curiosity. As this spans about 10 lines, it's pretty clear what `inputs_closed[0]` refers to

Reply:
> we would also support a min with a std::tuple right?

Of course we would!  It would be *super* embarrassing if I'd started by using a `std::array` here only to discover that `timpi/op_function.h` didn't support anything more complicated than `std::pair`, right?  And then I'd tried to slip a `std::vector` by everyone while I quietly worked on the TIMPI PR?

But anyway, we'll just stick with `vector` here for now, because ... reasons.

Reply:
Hmmm... actually, I've rediscovered the reason I didn't go crazy with `OpFunction` overloads earlier.  There's a bit of an ambiguity to some of them that isn't there for `StandardType` and `Packing` overloads, an it goes down a rabbit hole.

If I've declared a `tuple <T, U, V> x;` and I take a `min()` of those across two processors - should I get the component-wise min, or the dictionary-order `min()`?  If I do the former, won't that be a nasty surprise to people who instantiate `tuple <Year, Day, Month> iso_8601_date;`?  If I do the latter, won't that be a nasty surprise to people who instantiate `tuple <bool, bool, bool> packing_three_mins_into_one;`?  With a particular class it's maybe sufficient to pick one or the other and document it (e.g. our `OpFunction<Point>` says `min() and max() are applied component-wise; this makes them useful for bounding box reduction operations`, even though that makes them inconsistent with our `operator<(const Point&,const Point&)`, but who knows what someone expects from an arbitrary `tuple`?

I could create both types of functions, partially specialize `OpFunction` around a `componentwise_tuple<Types...>` and `dictionary_tuple<Types...>`, and then require people to use one or the other container (subclass? shim class??) when doing reductions?  Or I could create a `Communicator::allreduce()` that would take (and MPI_Op_create if necessary, etc) an explicit function rather than inferring a function via data type and communicator method name?  I could do both?

I'm leaning toward the latter at this point, but damn.  This is the sort of question that would have been ideal to bring up in person last week.

Reply:
Haha well you didn't make me think about this until this morning